jujugui/static/gui/src/app/components/deployment/choose-cloud/choose-cloud.js

```diff
@@ -22,18 +22,25 @@ YUI.add('deployment-choose-cloud', function() {
 
   juju.components.DeploymentChooseCloud = React.createClass({
 
-    // XXX huwshimi 22 March 2016: this should be replaced with a passed in prop
-    // once the data is available.
-    CREDENTIALS: [{
-      id: 'my-cloud-credentials',
-      cloud: 'aws',
-      title: 'My cloud credentials',
-      owner: 'Me!',
-      uses: 3
-    }],
-
     propTypes: {
-      changeState: React.PropTypes.func.isRequired
+      changeState: React.PropTypes.func.isRequired,
+      jem: React.PropTypes.object.isRequired
+    },
+
+    getInitialState: function() {
+      return {
+        credentials: []
+      };
+    },
+
+    componentWillMount: function() {
+      this.props.jem.listTemplates((error, data) => {
```

Comment:
Yeah, my initial thought was that some data munging was going to be needed to transform data into credentials, but that turned out to not be the case. Variable names updated.